src/main/commandServer/__tests__/settingsValidator.spec.ts

```diff
@@ -36,119 +53,359 @@ describe(SettingsValidator, () => {
-        kubernetes:
-          {
-            enabled:         desiredEnabledBoolean,
-            version:         '1.23.4',
-            containerEngine: 'moby'
+        if (props.length === 0) {
+          return;
+        }
+
+        describe.each(props.sort())(`${ prefix }%s`, (key) => {
+          const keyPath = path.concat(key);
+
+          if (keyPath.join('.') in platformSpecificFields) {
+            beforeEach(() => {
+              spyPlatform.mockReturnValue(platformSpecificFields[keyPath.join('.')]);
+            });
           }
-      });
-    });
 
-    it('should report errors for unchangeable fields', () => {
-      const desiredEnabled = !cfg.kubernetes.enabled;
-      const desiredEngine: settings.ContainerEngine =
-        cfg.kubernetes.containerEngine === settings.ContainerEngine.MOBY ? settings.ContainerEngine.CONTAINERD : settings.ContainerEngine.MOBY;
-      const requestedSettings = {
-        kubernetes:
-          {
-            enabled:                    desiredEnabled,
-            checkForExistingKimBuilder: !cfg.kubernetes.checkForExistingKimBuilder,
-            containerEngine:            desiredEngine,
+          it('should allow no change', () => {
+            const input = _.set({}, keyPath, _.get(cfg, keyPath));
+            const [needToUpdate, errors] = subject.validateSettings(cfg, input);
+
+            expect({ needToUpdate, errors }).toEqual({
+              needToUpdate: false,
+              errors:       [],
+            });
+          });
+
+          if (specialFields.some(specialField => _.isEqual(path.concat(key), specialField))) {
+            return;
           }
-      };
-      const [needToUpdate, errors] = subject.validateSettings(cfg, requestedSettings);
 
-      expect(needToUpdate).toBeFalsy();
-      expect(errors).toEqual(["Changing field kubernetes.checkForExistingKimBuilder via the API isn't supported."]);
-    });
+          it('should allow changing', () => {
+            let newValue: any;
 
-    it('should complain about all unchangeable fields', () => {
-      const valuesToChange: [RecursivePartial<settings.Settings>, string][] = [
-        [{ version: cfg.version + 1 }, 'version'],
-        [{ kubernetes: { memoryInGB: cfg.kubernetes.memoryInGB + 1 } }, 'kubernetes.memoryInGB'],
-        [{ kubernetes: { numberCPUs: cfg.kubernetes.numberCPUs + 1 } }, 'kubernetes.numberCPUs'],
-        [{ kubernetes: { port: cfg.kubernetes.port + 1 } }, 'kubernetes.port'],
-        [{ kubernetes: { checkForExistingKimBuilder: !cfg.kubernetes.checkForExistingKimBuilder } }, 'kubernetes.checkForExistingKimBuilder'],
-        [{ kubernetes: { WSLIntegrations: { stuff: 'here' } } }, 'kubernetes.WSLIntegrations'],
-        [{
-          kubernetes: {
-            WSLIntegrations: {
-              describe: true, three: false, keys: true
+            switch (typeof defaultSettings[key]) {
+            case 'boolean':
+              newValue = !_.get(cfg, keyPath);
```

Comment:
I like the use of `lodash.get`, but should we call these in a wrapper that does an `expect(v).toBeDefined()` before returning it?

Reply:
Looking at this, this is really dumb: `defaultSettings` is some child of `cfg` anyway, so this is just the same as `defaultSettings[key]` (and there's no point for `_.get()`, which also means that it can't be undefined, because we know it's a `boolean` here).  I'll make a commit to simplify.

Reply:
Fixed in 1d23d5ef7466246d567c9b3b316d07d4dff77107.